docs: adds an otelcollector example (#2277)
grafana/pyroscope: examples/otelcollector/README.md

> # Grafana Agent Pull Mode Integration

Reviewer: Changing title case to sentence case to match our style guide (see Writers' Toolkit). 
Keep in mind that we'll need to update this for Grafana Alloy, since Agent will be deprecated. 
```suggestion
# Grafana Agent mull mode integration
```

This example demonstrates how you can use Grafana Agent with Grafana Pyroscope to profile otel collector.

> ### 1. Configure Grafana agent

Reviewer: ```suggestion
### 1. Configure Grafana Alloy
```

> This example demonstrates how you can use Grafana Agent with Grafana Pyroscope to profile otel collector.
> 
> ### 1. Configure Grafana agent
> 

Reviewer: ```suggestion
This example demonstrates how you can use Grafana Alloy with Grafana Pyroscope to profile the OTel collector.

### 1. Configure Grafana Alloy

```

> For Grafana Alloy to pull profiles from your application, configure the agent for things like profiling configuration, targets, and possibly authentication.

Reviewer: Are you usingAlloy or Agent for this example? The links and text all point to Agent.

> You'll need to configure the Grafana agent for things like profiling configuration, targets, and possibly authentication in order to have the Grafana Agent pull profiles from your application.

Reviewer: ```suggestion
For Grafana Agent to pull profiles from your application, configure the agent for things like profiling configuration, targets, and possibly authentication.
```

> You can find a list of [arguments](https://grafana.com/docs/agent/next/flow/reference/components/pyroscope.scrape/#arguments) and [supported blocks](https://grafana.com/docs/agent/next/flow/reference/components/pyroscope.scrape/#blocks) in the [Grafana Agent documentation for pyroscope](https://grafana.com/docs/agent/next/flow/reference/components/pyroscope.scrape/)

Reviewer: The links in your content point to the /next branch. Should this be /next or /latest? 
```suggestion
You can find a list of [arguments](https://grafana.com/docs/agent/next/flow/reference/components/pyroscope.scrape/#arguments) and [supported blocks](https://grafana.com/docs/agent/next/flow/reference/components/pyroscope.scrape/#blocks) in the [Grafana Agent documentation for Pyroscope](https://grafana.com/docs/agent/next/flow/reference/components/pyroscope.scrape/)
```

Refer to [config file](./agent/config/config.river) to see an example of how to configure Grafana Agent to send profiling data to Pyroscope.

> ### 2. Run Grafana agent, Grafana and Pyroscope

Reviewer: Missing comma. 
```suggestion
### 2. Run Grafana agent, Grafana, and Pyroscope
```

> # Grafana Agent mull mode integration

Reviewer 2: ```suggestion
# Grafana Agent pull mode integration
```

This example demonstrates how you can use Grafana Agent with Grafana Pyroscope to profile otel collector.

### 1. Configure Grafana agent

> For Grafana Agent to pull profiles from your application, configure the agent for things like profiling configuration, targets, and possibly authentication.

Reviewer 2: ```suggestion
For Grafana Alloy to pull profiles from your application, configure the agent for things like profiling configuration, targets, and possibly authentication.
```
